dask/array/core.py

```diff
@@ -199,8 +199,14 @@ def zero_broadcast_dimensions(lol, nblocks):
     --------
     lol_tuples
     """
+    n = 0
```

Comment:
For clarity, consider keeping this logic to calculate `n` in separate helper function `_homogeneous_deepmap(func, seq)`.